Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp

```diff
@@ -1795,11 +1792,12 @@ bool AXIsolatedObject::isSelectedOptionActive() const
     return false;
 }
 
-bool AXIsolatedObject::hasMisspelling() const
+Vector<TextCheckingResult> AXIsolatedObject::spellCheckerResults() const
 {
-    return Accessibility::retrieveValueFromMainThread<bool>([this] () {
-        auto* axObject = associatedAXObject();
-        return axObject ? axObject->hasMisspelling() : false;
+    return Accessibility::retrieveValueFromMainThread<Vector<TextCheckingResult>>([this] () -> Vector<TextCheckingResult> {
+        if (auto* axObject = associatedAXObject())
+            return axObject->spellCheckerResults();
+        return { };
```

Comment:
We're returning a `Vector<TextCheckingResult>` from the main thread to the secondary thread. Is `TextCheckingResult` and all types within thread-safe?

Reply:
Changed it to returning and storing only the ranges instead of the whole TextSpellingResults. We should be ok as long as the ranges are not accessed in both threads after being created.